_pages/contributing/extensible-customizable-coding-guidelines.md

```diff
@@ -0,0 +1,39 @@
+---
+title: Writing Extensible and Customizable Code
+---
+
+To make Spartacus code easier to extend and customize, we should avoid using private constants for default values in the TypeScript class logic. Instead, those default values should be stored in `protected readonly` fields, unless it is justified to extract them as separate constants. For example, if might be justified when a `const` is used in multiple classes. In that case, the `const` should be exported to the public API using a clear, descriptive name.
```

Comment:
```suggestion
To make Spartacus code easier to extend and customize, we should avoid using private constants for default values in the TypeScript class logic. Instead, those default values should be stored in `protected readonly` fields, unless it is justified to extract them as separate constants. For example, it might be justified when a `const` is used in multiple classes. In that case, the `const` should be exported to the public API using a clear, descriptive name.
```

Reply:
Good catch! 👍 🙇